src/main/scala/com/workday/warp/dsl/WarpMatchers.scala

```diff
@@ -127,19 +131,20 @@ trait WarpMatchers {
     *
     * @param negated a [[ResultOfNotWordForAny]] representing a negated match condition.
     */
-  implicit class NegatedMatcher[T <: TrialResult[Any]](negated: ResultOfNotWordForAny[List[T]]) {
+  implicit class NegatedMatcher[T <: Iterable[TrialResult[_]]](negated: ResultOfNotWordForAny[T]) {
```

Comment:
let's talk about this change, I'd like to understand the motivation

Reply:
we are basically working around the lack of covariance on `ResultOfNotWordForAny`.

instead of accepting a `ResultOfNotWordForAny` of a `List` of any subtype of `TrialResult`,
we now accept a `ResultOfNotWordForAny` of any subtype of `Iterable[TrialResult]`.

(`Iterable` is covariant in its type parameter)